src/faqs/guidance-handler.js

```diff
@@ -0,0 +1,146 @@
+
+  // Validate presigned URL
+  if (!presignedUrl) {
+    log.error('No presigned URL provided in message data');
+    return badRequest('Presigned URL is required');
+  }
+
+  const site = await Site.findById(siteId);
+  if (!site) {
+    log.error(`Site not found for siteId: ${siteId}`);
+    return notFound('Site not found');
+  }
+
+  try {
+    // Fetch FAQ data from presigned URL
+    log.info(`Fetching FAQ data from presigned URL: ${presignedUrl}`);
+    const response = await fetch(presignedUrl);
+
+    if (!response.ok) {
+      log.error(`Failed to fetch FAQ data: ${response.status} ${response.statusText}`);
+      return badRequest(`Failed to fetch FAQ data: ${response.statusText}`);
+    }
+
+    const faqData = await response.json();
+
+    const { url, faqs } = faqData;
+
+    // Validate the fetched data
+    if (!faqs || !Array.isArray(faqs) || faqs.length === 0) {
+      log.info('No FAQs found in the response');
+      return noContent();
+    }
+    log.info(`Received FAQ data with ${faqs.length} FAQ topics`);
+
+    // Filter to count only suitable suggestions
+    const totalSuitableSuggestions = faqs.reduce((count, faq) => {
+      const suitable = (faq.suggestions || []).filter(
+        (s) => s.is_answer_suitable && s.is_question_relevant,
+      );
+      return count + suitable.length;
+    }, 0);
+
+    if (totalSuitableSuggestions === 0) {
+      log.info('No suitable FAQ suggestions found after filtering');
+      return noContent();
+    }
+
+    // Create guidance object (similar to summarization)
+    const wrappedGuidance = {
+      recommendations: [{
+        insight: `${totalSuitableSuggestions} relevant FAQs identified based on top user prompts in your brand presence analysis`,
+        rationale: 'When your content aligns with the user intent recognized by large language models (LLMs), it becomes easier for these models to reference or mention your page in their responses',
+        recommendation: 'Add the relevant FAQs listed below to the corresponding pages',
+        type: 'CONTENT_UPDATE',
+      }],
+    };
+
+    // Find or create opportunity
+    const existingOpportunities = await Opportunity.allBySiteId(siteId);
+    let opportunity = existingOpportunities.find(
+      (oppty) => oppty.getData()?.subType === 'faqs',
+    );
+
+    const entity = createOpportunityData(siteId, auditId, url, wrappedGuidance);
+
+    if (!opportunity) {
+      opportunity = await Opportunity.create(entity);
+      log.info(`Created new FAQ opportunity: ${opportunity.getId()}`);
+    } else {
+      opportunity.setAuditId(auditId);
+      opportunity.setData({
+        ...opportunity.getData(),
+        ...entity.data,
+      });
+      opportunity.setGuidance(wrappedGuidance);
+      opportunity.setUpdatedBy('system');
+      opportunity = await opportunity.save();
+      log.info(`Updated existing FAQ opportunity: ${opportunity.getId()}`);
+    }
```

Comment:
Not opposed to this, but curious if you have a specific reason for not using `convertToOpportunity`?

Reply:
`convertToOpportunity` only works for updating oppties of specific types (like `structured-data`)
as this audit produces a generic oppty, we're using a `subType` field in order to differentiate other oppties of the same kind (e.g. summarization)
there are custom logics here than the default in `convertToOpportunity` to look up the audit by sub type and update.